README.md

```diff
@@ -7,8 +7,58 @@ Primarily, it allows reading a service description and generating various
 types of output.  It uses the [Jinja2 templating language](https://jinja.palletsprojects.com) ([(alt. link)](https://jinja2docs.readthedocs.io/en/stable/))
 for most output definitions.
 
-# Testing it out
+## Getting started
 
+### Prerequisites
+* Python 3.8 installed
+* If the installation (pip install) is executed behind a (corporate) proxy, the following environments variables must be set: `http_proxy` and `https_proxy` (including authentication e.g., `http://${proxy_username):$(proxy_password)@yourproxy.yourdomain`)
+* If you do not run with administration rights, you may need to configure pip target path to write to your user home directory or consider [using the `pipenv` method](#setup-with-pipenv).
+
+```
+On Unix and Mac OS X the configuration file is: $HOME/.pip/pip.conf
+If the file does not exist, create an empty file with that name.
+
+Add or replace the following lines:
+[global]
+target=/somedir/where/your/account/can/write/to
+
+On Windows, the configuration file is: %HOME%\pip\pip.ini
+If the file does not exist, create an empty file with that name.
+
+Add or replace the following lines:
+[global]
+target=C:\SomeDir\Where\Your\Account\Can\Write\To
+```
+### Project Setup
+
+* If you use a custom pip installation directory, set the `PYTHONPATH` environment variable to the directory that you set in the `pip.ini` file.
+
+### Setup with `pipenv`
+[pipenv](https://pypi.org/project/pipenv/) is a tool that manages a virtual environment and install the package and its dependencies, making the process much simpler and predictable, since the `Pipfile` states the dependencies, while `Pipfile.lock` freezes the exact version in use.
+
+If [`pyenv` shell command](https://github.com/pyenv/pyenv) is not installed, use its [installer](https://github.com/pyenv/pyenv-installer) to get it:
+
+```sh
+curl https://pyenv.run | bash  # download and install
+exec $SHELL                    # restart your shell using the new $PATH
+```
+
+Make sure Python version 3.8.5 is installed:
```

Comment:
Does it necessarily have to be 3.8.5, or would any 3.8.X-version do? If it is not strict we could add something like `(or any later 3.8.X version, if available)`.

I am no Pip expert , but the Pipfile says ` python_version = "3.8"` so I assume that any 3.8.X version would suffice, or?

Reply:
Hi @erikbosch,

To be honest about python version, I've take the same version that one used on vss-tools and made the test with that one.
But you're right, there is an inconsistency between Pipfile and documentation.

By having a look to python 3.8 series changelog [here](https://docs.python.org/3.8/whatsnew/changelog.html#python-3-8-12), it seems changes between each releases is only about bugfix or security issues.

I've also made a test with python 3.8.0 and python 3.8.12. Both are working properly (I haven't tried each versions between 😉). Moreover, when you are using _pyenv_ and _pipenv_ together and that Pipfile mentions only 3.8, _pyenv_ will automatically download the latest python version available for this minor version (for 3.8, it is 3.8.12 actually).

If everyone agree, I can adapt the documentation to indicate that python version of 3.8.x series will do the job in any cases.